segment_vector.go

```diff
@@ -58,7 +58,9 @@ type VecPostingsIterator interface {
 
 type VectorIndex interface {
 	// @params: Search params for backing vector index (like IVF, HNSW, etc.)
-	Search(qVector []float32, k int64, params json.RawMessage) (VecPostingsList, error)
+	// @requiresFiltering: whether the kNN queries require any filtering.
+	Search(qVector []float32, k int64, requiresFiltering bool,
+		eligibleDocIDs *roaring.Bitmap, params json.RawMessage) (VecPostingsList, error)
```

Comment:
Don't quite see the need for `requiresFiltering` here either. Can you not check `eligibleDocIDs.Size() > 0` to make any decision here. A separate API - `SearchWithFilter` feels a bit more tasteful.

Reply:
RequiresFiltering indicates whether or not the kNN query is a filtered kNN query.
For a non-filtered kNN query, eligible doc IDs should always be nil since there's no concept of an 'include' list over there, in line with existing behaviour. 

In case of a match none filter, for example, the count of eligible docIDs will be 0. Hence, eligible docIDs will be 0 but this will not be an unfiltered query.  This is true for any filter which has no eligible docs in the segment.

With that background, 'requiresFiltering' was to distinguish between filtered and unfiltered to handle unfiltered kNN without any change.

Reply:
I shall work on handling this at the bleve layer instead of the zapx layer, like in my current approach.

Reply:
Added the search with filter function in the latest commit.